server/lib/src/idm/server.rs

```diff
@@ -1934,6 +1935,7 @@ impl IdmServerProxyWriteTransaction<'_> {
         let modlist = ModifyList::new_list(vec![
             // Ensure that the account has no validity, and the expiry is now.
             m_purge(Attribute::AccountValidFrom),
+            m_purge(Attribute::AccountExpire),
             Modify::Present(Attribute::AccountExpire, v_expire),
```

Comment:
It seems silly to be removing then setting expiry here?

Reply:
It's not silly - the modifies are stateful. If you set modify::present, it means "make sure that value exists". It DOESN'T clear existing values. That's why you need to remove then set.

Reply:
Yep, it's just saying "assert that a value exists in the array" basicly. It doesn't remove anything already there.

Reply:
We do have an upsert option in Modify but I didn't use it here.

Reply:
Thanks for explaining it